assets/js/util/convert-array-to-keyed-object-map.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * convertArrayListToKeyedObjectMap utility function.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * Converts an array of objects into an object map containing objects.
+ *
+ * @param {Array} arrayData       Array data to be converted to an Object
+ * @param {string} propertyKey  Object propert to be used for the property name in the main object.
+ *
+ * @return {Object} An object containing objects
+ */
+export function convertArrayListToKeyedObjectMap( arrayData, propertyKey ) {
```

Comment:
This is nice, but doesn't lodash already have a utility like this? I'm usually a fan of using straightforward ES6 functions like this, but if we _already_ include lodash (and it doesn't look like we'll be removing it anytime soon), we can use that instead.

i think https://lodash.com/docs/4.17.15#keyBy does it. If we can use that, let's do that instead. But this looks fine if there's a reason we need to use it/can't use lodash. 🙂